commands/agentupgrade_script.go

```diff
@@ -186,21 +236,128 @@ def update_unit_config(agent, data):
 
     return data
 
+def update_init_scripts(series, lxc_agent, lxd_agent):
+    if series == 'trusty':
+        update_upstart_scripts(lxc_agent, lxd_agent)
+    else:
+        update_systemd_scripts(lxc_agent, lxd_agent)
+
+def update_upstart_scripts(lxc_agent, lxd_agent):
+    lxc_path = path.join(UPSTART_DIR, upstart_conf(lxc_agent))
+    lxd_path = path.join(UPSTART_DIR, upstart_conf(lxd_agent))
+    rewrite_lxc_to_lxd(lxc_path, lxd_path)
+    safe_unlink(lxc_path)
+
+def rewrite_lxc_to_lxd(lxc_path, lxd_path):
+    "Copies contents of lxc_path file into lxd_path, converting lxc->lxd on the way"
+    with open(lxc_path, 'r') as source:
+        data = source.read()
+    updated = data.replace('lxc', 'lxd')
+    with open(lxd_path, 'w') as dest:
+        dest.write(updated)
+
+def update_systemd_scripts(lxc_agent, lxd_agent):
+    lxc_dir = path.join(INIT_DIR, 'jujud-' + lxc_agent)
+    lxd_dir = path.join(INIT_DIR, 'jujud-' + lxd_agent)
+
+    # Create lxd versions of service file and exec-start script.
+    os.mkdir(lxd_dir)
+    lxd_service_path = path.join(lxd_dir, systemd_conf(lxd_agent))
+    rewrite_lxc_to_lxd(
+        path.join(lxc_dir, systemd_conf(lxc_agent)),
+        lxd_service_path)
+    lxd_exec_start = path.join(lxd_dir, 'exec-start.sh')
+    rewrite_lxc_to_lxd(path.join(lxc_dir, 'exec-start.sh'), lxd_exec_start)
+    os.chmod(lxd_exec_start, 0o755)
+
+    shutil.rmtree(lxc_dir)
+
+    # Correct the link from /etc/systemd/system
+    os.unlink(path.join(SYSTEMD_DIR, systemd_conf(lxc_agent)))
+    force_symlink(lxd_service_path, path.join(SYSTEMD_DIR, systemd_conf(lxd_agent)))
+
+def get_series():
```

Comment:
why not just use the series.HostSeries from utils/series?

Reply:
It seemed simpler to just work this out on the machine than to work it out in Go and then pass it to the Python from both `upgrade-agents` and `abort`.